HorizonEngine/src/HorizonEngine/Components/Component.h

```diff
@@ -234,34 +234,94 @@ namespace Hzn
 		RenderComponent(const glm::vec4& color) : m_Color(color) {};
 		~RenderComponent() = default;
 		// conversion operator for render component.
-		operator const glm::vec4() const & { return m_Color; }
-		operator glm::vec4() & { return m_Color; }
+		operator const glm::vec4() const& { return m_Color; }
+		operator glm::vec4()& { return m_Color; }
 
 		glm::vec4 m_Color = glm::vec4(1.0f);
 
+		std::shared_ptr<Hzn::Texture2D> m_Texture;
```

Comment:
I don't think you need to have a pointer to the texture.

Reply:
yes. you are right. we only neeed to store the path